### ISSUE TITLE
Write a tutorial, linked from Bazel front page

### ISSUE THREAD

**Issue by jayconrod** [bazelbuild/rules_go#637]
https://bazel.build/ has tutorials for Java, C++, iOS, and Android, linked from the bottom of the home page. Each tutorial has language-specific setup instructions, but also covers general Bazel concepts (WORKSPACE, BUILD files, rules, etc).

We should have a tutorial for Go. This is especially important since we have some Go-specific steps (WORKSPACE boilerplate, running Gazelle, etc).

It probably doesn't make sense to write this today, since the API and Gazelle are changing quickly, but this should be written soon, certainly by the end of 2017Q3.

**Comment by alexeagle** [comment]
@chrislovecnm wrote a Go tutorial: https://bazel-contrib.github.io/SIG-rules-authors/go-tutorial.html

I think the remaining ask here is just to get that linked from bazel.build.

**Comment by achew22** [comment]
Generally speaking I'd be happy to link to documentation. One concern I have with it is that it recommends some advice that is explicitly [disavowed](https://github.com/golang-standards/project-layout/issues/117) by the Go team. Someone poached a high SEO name on GitHub and has been giving out that bad advice for years and I'm not sure I want to continue to propagate something that's disfavored

**Comment by chrislovecnm** [comment]
@achew22, you are concerned about the use of a directory named pkg?  Just clarifying.

**Comment by achew22** [comment]
Yep, that's the big one. I think if you were to rename `pkg` to `internal` it might also serve as a nice demo of package visibility automation in Gazelle as well
